docs/DeveloperGuide.adoc

```diff
@@ -295,56 +301,479 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+** Use case resumes from step 4
+
+--
+
+[none]
+* 5a. No text entered
++
+[none]
+** 5a1. NoteSys shows a warning message
+
++
+Use case resumes from step 5
+
+[discrete]
+=== UC32 - Delete a note
+
+*MSS*
+
+1. User chooses to delete a note
+2. User enters description of note
+3. NoteSys removes the note
+
++
+Use case ends
+
+*Extensions*
+[none]
+* 2a. Description entered is invalid
++
+
++
+[none]
+** 2a1. NoteSys shows an error message
++
+Use case ends
+
+[discrete]
+=== UC33 - Update a note
+
+*MSS*
+
+1. User chooses to update a note
+2. User enters description of password
+3. NoteSys opens the note
+4. User edits note details
+5. User submits the note
+6. NoteSys saves the note
+
++
+Use case ends
+
+*Extensions*
+[none]
+* 2a. Description entered is invalid
++
+
++
+[none]
+** 2a1. NoteSys shows an error message
++
+Use case ends
+[none]
+* 5a. No text entered
++
+
++
+[none]
+** 5a1. NoteSys shows an error message
++
+Use case ends
+
+---
+
+System: `CardSys`
+[discrete]
+=== Use Case:
+1. UC41 - Add a card
+2. UC42 - Delete a card
+
+---
+
+[discrete]
+=== UC41 - Add a card
+
+*MSS*
+
+1. User chooses to add a card
+2. User enters description and details
+3. CardSys adds the card
+
++
+Use case ends
+
+*Extensions*
+[none]
+* 2a. Details entered are invalid
++
+
++
+[none]
+** 2a1. CardSys shows an error message
+** 2a2.CardSys requests for new details
+** 2a3. User enters new details
+** Steps 2s1-2s3 are repeated until the details entered are valid
+** Use case resumes from step 3
+
+[discrete]
+=== UC42 - Delete a card
+
+*MSS*
+
+1. User chooses to delete a card
+2. User enters description
+3. CardSys removes the card
+
++
+Use case ends
+
+*Extensions*
+[none]
+* 2a. Description entered does not exist
++
+
++
+[none]
+** 2a1. CardSys shows an error message
++
+Use case ends
 
 [appendix]
 == Non Functional Requirements
+Accessibility
+
+* A user shall be able to download the released JAR file conveniently from the newest tagged release.
+*  The app shall be accessible by anyone who has downloaded the released JAR file.
+* Should be accommodating for both advanced, seasoned users as well as new users.
+
+
+Efficiency
+
+*  The response of the app to any user action shall appear within 5 seconds.
+
+Performance
+
+*  The app shall be able to contain up to 1000 items without any drop in performance.
+
+Reliability
+
+* The app shall throw appropriate exceptions when any user input is invalid or any user action fails to execute completely.
+
+Security
+
+* The app shall resist unauthorised, accidential or unintended usage and provide access only to legitimate users.
+
+Usability
+
+*  A user with above average typing speed for regular English (i.e. not code, not system admin commands) shall be able to accomplish most of the tasks faster using commands than using the mouse.
+
+Data Integrity
+
+* Should be able to check for the data integrity as to verify that no one has modified the files within secureIT in an unauthorised fashion.
```

Comment:
This sounds too functional, maybe it would be better to rephrase the sentence into `this program should consider data integrity of files`.